user-data.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# This file is templated by Terraform's templatefile function.
+# Variables written like (without spaces): $ { var } will be interpreted by
+# Terraform, NOT the shell. To write a variable for the shell to interpret,
+# use $var.
+
+spacelift () {(
+set -e
+
+${TF_CONFIGURATION}
+
+cw_config_ver=${TF_CW_CONFIG_VER}
+if [[ $cw_config_ver != 0 ]];then
+  echo "Configuring CloudWatch Agent with config version ${cw_config_ver}"
+  cw_config_source=/tmp/amazon-cloudwatch-agent-v${cw_config_ver}.json
+  cw_config_dest=/opt/aws/amazon-cloudwatch-agent/bin/config.json
+  cp "${cw_config_source}" ${cw_config_dest}
+  /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c file:${cw_config_dest}
```

Comment:
I think if we can use my suggestion of just copying both config files into the cloudwatch agent directory in the image, we can remove all the copies from here and simplify it down to a one-liner:

```shell
/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c file:/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-v${cw_config_ver}.json
```